test/e2e/specs/status-page.spec.js

```diff
@@ -0,0 +1,129 @@
+        await page.getByTestId("custom-css-input").getByTestId("textarea").fill(customCss); // Prism
+        await expect(page.getByTestId("description-editable")).toHaveText(descriptionText);
+        await expect(page.getByTestId("custom-footer-editable")).toHaveText(footerText);
+
+        // Add an incident
+        await page.getByTestId("create-incident-button").click();
+        await page.getByTestId("incident-title").isEditable();
+        await page.getByTestId("incident-title").fill(incidentTitle);
+        await page.getByTestId("incident-content-editable").fill(incidentContent);
+        await page.getByTestId("post-incident-button").click();
+
+        // Add a group
+        await page.getByTestId("add-group-button").click();
+        await page.getByTestId("group-name").isEditable();
+        await page.getByTestId("group-name").fill(groupName);
+
+        // Add the monitor
+        await page.getByTestId("monitor-select").click(); // Vue-Multiselect component
+        await page.getByTestId("monitor-select").getByRole("option", { name: monitorName }).click();
+        await expect(page.getByTestId("monitor")).toHaveCount(1);
+        await expect(page.getByTestId("monitor-name")).toContainText(monitorName);
+
+        // Save the changes
+        await screenshot(testInfo, page);
+        await page.getByTestId("save-button").click();
+        await expect(page.getByTestId("edit-sidebar")).toHaveCount(0);
+
+        // Ensure changes are visible
+        await expect(page.getByTestId("incident")).toHaveCount(1);
+        await expect(page.getByTestId("incident-title")).toContainText(incidentTitle);
+        await expect(page.getByTestId("incident-content")).toContainText(incidentContent);
+        await expect(page.getByTestId("description")).toContainText(descriptionText);
+        await expect(page.getByTestId("group-name")).toContainText(groupName);
+        await expect(page.getByTestId("footer-text")).toContainText(footerText);
+        await expect(page.getByTestId("powered-by")).toHaveCount(0);
+
+        await expect(page.getByTestId("update-countdown-text")).toContainText("00:");
+        const updateCountdown = Number((await page.getByTestId("update-countdown-text").textContent()).match(/(\d+):(\d+)/)[2]) ;
+        expect(updateCountdown).toBeGreaterThanOrEqual(refreshInterval); // cant be certain when the timer will start, so ensure it's within expected range
+        expect(updateCountdown).toBeLessThanOrEqual(refreshInterval + 10);
+
+        await expect(page.locator("body")).toHaveClass(theme);
+        expect(await page.locator("head").innerHTML()).toContain(googleAnalyticsId);
+
+        const backgroundColor = await page.evaluate(() => window.getComputedStyle(document.body).backgroundColor);
+        expect(backgroundColor).toEqual("rgb(0, 128, 128)");
+
+        await screenshot(testInfo, page);
+
+        // Flip the "Show Tags" and "Show Powered By" switches:
```

Comment:
I am not a testing expert.
Do you think it could make sense to move the setup code above to `beforeEach`?

Reply:
I think it would be better to do when & if more cases are added to the spec.